tests/utils.ts

```diff
@@ -37,10 +38,44 @@ export const test = baseTest.extend({
   },
 })
 
-export function getYesterday(): string {
-  const yesterday = moment().subtract(1, 'days')
-  const formattedDate = yesterday.format('DD/MM/YYYY')
-  return formattedDate
+export function getPastDate(): Date {
+  return moment('2024-02-12 15:00:00').toDate()
+}
+
+export async function setBrowserTime(date: Date, page: Page | BrowserContext) {
+  const fakeNow = date.valueOf()
+
+  // Update the Date accordingly
+  await page.addInitScript(`{
+    // Extend Date constructor to default to fakeNow
+    Date = class extends Date {
+      constructor(...args) {
+        if (args.length === 0) {
+          super(${fakeNow});
+        } else {
+          super(...args);
+        }
+      }
+    }
+    // Override Date.now() to start from fakeNow
+    const __DateNowOffset = ${fakeNow} - Date.now();
+    const __DateNow = Date.now;
+    Date.now = () => __DateNow() + __DateNowOffset;
+  }`)
+}
+
+export function urlMatcher(): Matcher {
+  return customMatcher({
+    urlComparator(a, b) {
+      const fieldsToRemove = ['t', 'date_from', 'date_to']
+      ;[a, b] = [a, b].map((url) => {
+        const urlObj = new URL(url)
+        fieldsToRemove.forEach((field) => urlObj.searchParams.delete(field))
+        return urlObj.toString()
+      })
+      return a === b
+    },
+  })
 }
```

Comment:
just a suggestion - you can create a single instance of the `urlMatcher`

```suggestion
export const urlMatcher: Matcher = customMatcher({
    urlComparator(a, b) {
      const fieldsToRemove = ['t', 'date_from', 'date_to']
      ;[a, b] = [a, b].map((url) => {
        const urlObj = new URL(url)
        fieldsToRemove.forEach((field) => urlObj.searchParams.delete(field))
        return urlObj.toString()
      })
      return a === b
    },
  })
```

And on usage:
```
advancedRouteFromHAR('tests/HAR/clearbutton.har', {
      updateContent: 'embed',
      update: false,
      notFound: 'abort',
      url: /stride-api/,
      matcher: urlMatcher, // NOTE - no `()` are needed
})
```

Reply:
updated, thanks!